src/kernel/manager.ts

```diff
@@ -81,13 +93,14 @@ class KernelManager implements Kernel.IManager {
    *
    * @param options - Overrides for the default options.
    */
-  listRunning(options?: Kernel.IOptions): Promise<Kernel.IModel[]> {
-    return Kernel.listRunning(this._getOptions(options)).then(running => {
+  listRunning(options?: Kernel.IOptions): Promise<ISequence<Kernel.IModel>> {
+    return Kernel.listRunning(this._getOptions(options)).then(it => {
+      let running = toArray(it);
       if (!deepEqual(running, this._running)) {
-        this._running = running.slice();
-        this.runningChanged.emit(running);
+        this._running = running;
+        this.runningChanged.emit(new Vector(this._running));
       }
-      return running;
+      return iter(running);
```

Comment:
Can return `it` instead of `iter(running)` or is there a mutability issue there?